### DISCUSSION TITLE
Conflict when running "read -e"??

### DISCUSSION THREAD

**Discussion by psymole** [akinomyoga/ble.sh#528] [Q&A] [answered]
Hi!

I just recently discover this project, so please forgive me if I'm missing something obvious. 

In a few scripts I use `read -e -r -p "Select a file: " file_selected`.
Adding the `-e` flag allow the user to perform Tab Completion on the files present on the directory. 

If I run the same command while on a blesh session it doesn't perform tab completion and the variable is not set either. 

Is there something I can configure on blesh to have it work properly on these cases?

Thanks. 

P.S: Happy to provide more details if needed.
Same issue happens on Linux and Mac OS

**Comment by akinomyoga** [comment]
It's just because the completion is disabled in `read -e`.

You can configure it in the `read` keymap, (though the completion menu is not displayed):

```bash
# blerc

function ble/complete/context:plain-filename/generate-sources { sources=('file 0'); }
ble-bind -m read -f C-i 'complete context=plain-filename'
```

**Reply by akinomyoga** [reply]
Maybe I'll make this behavior default, but I need time to think about it.

**Reply by psymole** [reply]
Thanks for answering so quickly!

The config you propose got me to 99% of what I was looking to achieve. However, after setting the options on my blerc, I get the completion but the `-p` prompt does not work.

<img width="519" alt="image" src="https://github.com/user-attachments/assets/2fd4e2e8-cb8f-4a57-933f-dd15c61e269d">

The completion is working but the select file prompt does not.

Again thanks for a fabulous piece of software.

**Reply by akinomyoga** [reply]
Thanks for the additional report. However, the problem doesn't seem to reproduce in my environment.

Do you have any specific settings that cause this behavior? For example, does the problem reproduce with ble.sh plus plain Bash (without any settings other than https://github.com/akinomyoga/ble.sh/discussions/528#discussioncomment-11233513)? If so, could you identify what would cause the problem?

Also, could you provide the output of the following command?

```bash
$ ble/widget/display-shell-version
```

**Reply by psymole** [reply]
Same issue with as clean of a bash setup as I could think to do.

<img width="742" alt="image" src="https://github.com/user-attachments/assets/3cc49bf1-4c1b-4334-99db-3c8b842b11fb">

Here's the Info
```
M52WH6FHVW:~ alex.trujillo$ ble/widget/display-shell-version
GNU bash, version 5.2.37(1)-release (aarch64-apple-darwin23.4.0)
ble.sh, version 0.4.0-nightly+31f264a (noarch) [git 2.46.2, GNU Make 4.3, GNU Awk 5.1.0, API: 3.0 (GNU MPFR 4.1.0, GNU MP 6.2.1)]
locale: LANG=en_US.UTF-8
terminal: TERM=xterm-256color wcwidth=15.1-west/16.0-2+ri, xterm:95 (1;95;0)
options: +inherit_errexit +login_shell
```
I emptied out .bash_profile there's no .bashrc for my config

```
# ~/.bash_profile

# https://github.com/akinomyoga/ble.sh
source ~/.local/share/blesh/ble.sh
```
Here's the .bleshrc

```
# blerc

echo "test"

function ble/complete/context:plain-filename/generate-sources { sources=('file 0'); }
ble-bind -m read -f C-i 'complete context=plain-filename'
# ble-bind -m read -f C-i 'echo this'
```

**Reply by akinomyoga** [reply, marked as answer]
Hmm, the problem doesn't reproduce in my environment (Linux and Cygwin). I need to think about possible conditions later. If you find something until then, please let me know.

![image](https://github.com/user-attachments/assets/0caa5abc-6576-41c4-8539-2e82287c2d67)
![image](https://github.com/user-attachments/assets/37efd85c-31fd-49a4-8383-563587fca47b)

```
GNU bash, version 5.2.26(1)-release (x86_64-redhat-linux-gnu) [Fedora Linux 39 (Server Edition)]
ble.sh, version 0.4.0-devel4+31f264ad (noarch) [git 2.46.0, GNU Make 4.4.1, GNU Awk 5.2.2, API 3.2, PMA Avon 8-g1, (GNU MPFR 4.2.0-p12, GNU MP 6.2.1)]
locale: LANG=ja_JP.UTF-8
terminal: TERM=screen.xterm-256color wcwidth=16.0-emacs/16.0-2+ri, screen:49900 (83;49900;0), contra:0 (99;0)
options: +inherit_errexit

GNU bash, version 5.2.21(1)-release (x86_64-pc-cygwin)
ble.sh, version 0.4.0-devel4+31f264ad (noarch) [git 2.45.1, GNU Make 4.4.1, GNU Awk 5.3.0, API 4.0, PMA Avon 8-g1, (GNU MPFR 4.2.1, GNU MP 6.3.0)]
locale: LANG=ja_JP.UTF-8
terminal: TERM=screen.xterm-256color wcwidth=16.0-emacs/16.0-2+ri, screen:49900 (83;49900;0), contra:0 (99;0)
options: +inherit_errexit
```

**Reply by akinomyoga** [reply]
@psymole After that, I could actually reproduce the behavior with a slightly different setup. It is probably caused by the same issue, so I'll try to fix it later when I have time. I'm currently busy, so I might not be able to find the time for a few weeks, though. Thanks for your patience.
